### PR TITLE
tests: tag spread tests with basic tag on core test suite

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The idea of this change is to tag tests, those tags can be used then by
spread to define which tests have to be executed.

This first change includes the basic tests, which cover the basic
functionalities of snapd. Next step is to define which tests cover
alternative paths of the features, which cover errors, which are
regressions, etc.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM!

File: tests/core/apt/task.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ summary: Ensure that the core systems have a special apt placeholder
 
 systems: [ubuntu-core-*]
 
+tags: [ basic ]
```
Comment (pedronis) [inline tests/core/apt/task.yaml:5]: this seems a test that is enough to run with PRs, it involves something that shouldn't change often at all and isn't really related to hardware or kernel

File: tests/core/classic-snap16/task.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ summary: Ensure classic dimension works correctly
 # recommended instead of the classic snap
 systems: [ubuntu-core-16-*]
 
+tags: [ basic ]
```
Comment (pedronis) [inline tests/core/classic-snap16/task.yaml:7]: this one seems also fine to just run with PRs

File: HACKING.md [hunk 3]
```diff
@@ -548,6 +548,15 @@ The nested suites use some environment variables to configure the suite and the
     NESTED_CUSTOM_IMAGE_URL: Download and use an custom image from this url
 
 
+### Tagging spread tests
+
+Tags are used to categorize spread tests independently of the test suite where it is located.
```
Comment (pedronis) [inline HACKING.md:553]: I think the subclause should be: "where they are located."  as the main clause says "tests"

**Review by pedronis — COMMENTED** [review]
thank you, I commented on some tests that I think can be left out. OTOH I think some of the followings should be marked basic, as they involve the kernel or reboots or interaction with other components:

tests/core/enable-disable-units-gpio/task.yaml
tests/core/kernel-snap-refresh-on-core/task.yaml
tests/core/kernel-ver/task.yaml  (this one is not fundamental but is very cheap and is a consistency check)

and possibly

tests/core/netplan-cfg/task.yaml

I notice that
tests/core/gadget-kernel-refs-update-pc/task.yaml
is described as artificial but there's a TODO that we should get a real pi test that makes sense for validation for it.

File: tests/core/apt/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/core/apt/task.yaml:5]: sure, removed

File: tests/core/classic-snap16/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/core/classic-snap16/task.yaml:7]: removed

File: HACKING.md (hunk 3 above)
Comment (sergiocazzolato) [reply HACKING.md:553]: updated

**Comment by sergiocazzolato** [comment]
@pedronis hi, I updated the tests based on your comments. Thanks for the review.

About the TODO in gadget-kernel-refs-update-pc, I'll create a PR this week with a new test for arm devices.

**Review by pedronis — APPROVED** [review]
thank you

**Comment by anonymouse64** [comment]
Unfortunately I don't think I will be able to get to this, but in any case it has a conflict in the tests/utils dir

**Pull request closed without merging**